### ISSUE TITLE
Problems with: "Y" coordinates and Subdivisions.

### ISSUE THREAD

**Issue by GamerElit** [GriefPrevention/GriefPrevention#1294]
The "Y" coordinates are not protected as they should be.
In addition, the subdivision does not work well either.

I thought that the operation of the protection should protect the entire "Y" coordinate. That is to say that it should protect all the blocks from top to bottom (up to the bedrock).
And the issue of subdivisions, if a user has permissions (From "/trust") in a subdivision should be able to modify the terrain, chests, etc.. But this does not happen.

Y" coordinates that are not protected:
It just happens all the time, mostly when a protection is extended.
The coordinates above the chest are well protected, but those below are not, and any user can interact with that part. Also, the blocks below the chest are not shown as protected.
Subdivisions:
1.- The owner of a protection gives permissions to a user ("/trust").
2.- Then he subdivides it, in case the user does not have permissions in that subdivision, he gives permissions again ("/trust").
3.- For a while, it works fine. Then, it stops working, simply the user that had permissions can not do anything within that same subdivision (And in the list of "/trustlist" if it appears as having permissions).
4.- For a moment it works fine again, but it starts to work badly in other aspects. Certain blocks cannot be placed and then the whole protection starts to work like this.

`/version`: **This server is running Tuinity version git-Tuinity-"7734ef0" (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)**
`/version GriefPrevention`: **GriefPrevention version 16.17.1-2b69191**

No errors are generated, there are no logs of anything happening. The timings show everything fine.

### GriefPrevention config.yml

```yml

# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/
GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
    world_void: -1
  Claims:
    Mode:
      world_the_end: Disabled
      world_void: Disabled
      world: Survival
      world_nether: Disabled
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 50
    Max Accrued Claim Blocks:
      Default: 500000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 90
        ExceptWhenOwnerHasTotalClaimBlocks: 100000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: true
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local;/.
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg;/.
    WarningMessage: Por favor reduce tu velocidad. ¡No nos gustan los spamers!
    BanOffenders: true
    BanMessage: Baneado por spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
      world_void: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: true
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa;/warp;/warps
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 25.0
    ClaimBlocksSellValue: 0.25
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: true
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 3
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: true
  BanCommandPattern: Kick %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false

```

Plugins (32): 40ServidoresMC, AdvancedRegionMarket, AuthMe, BloodMoon, ChestShop, Chunky, DivineDrop, DropHeads, Essentials, EssentialsChat, EssentialsSpawn, ExploitFixer, FastAsyncWorldEdit (WorldEdit), GriefPrevention, GSit, HamsterAPI*, IllegalStack, LuckPerms, Multiverse-Core, ProtocolLib, SimpleClans, SkinsRestorer, spark, TAB, TabCompleteFilter, TreeAssist, Vault, VoidGenerator, Votifier, WorldBorder, WorldGuard

With the theme of the steps. Yes, it still happens but only if there are many protections or very large protections.

**Comment by RoboMWM** [comment]
I'll await your debugging steps that you deleted.

As well as the reproduction steps, because I can't really follow the situation you explained. Much easier and less prone to assumptions/variables when you can describe a simplified reproduction scenario

**Comment by GamerElit** [comment]
(Y coordinates)
1.- A user places a chest and protects a zone or A user places the command "/claim" and protects a zone.
2.- The zone is fully protected in the whole Y-coordinate.
3.- The owner of the zone extends its protection by many blocks. And now only the upper part of the protection is being protected, i.e. from a certain point the "Y" coordinate is no longer protected.
(This also happens when:)
1.- A user places a chest and protects a zone or A user places the command "/claim" and protects a zone.
2.- The zone is fully protected in the whole Y-coordinate.
3.- The owner of the zone creates multiple protected zones. And now only the upper part of the protection is being protected, i.e. from a certain point the "Y" coordinate is no longer protected.

(Where is that certain point when it stops protecting?)
It is 1 or 2 blocks below where the user placed the protection. That is to say that if the user placed the command "/claim" at coordinate 0 everything below coordinate -2 is not protected.

I don't quite understand the last part, but as I said. I have tried everything below and still the same error:
- [✔] I attempted running only GriefPrevention on the server
- [✔] Issue still occurs even if GriefPrevention is the only plugin running
- [✔] I attempted testing for the issue on a new server
- [✔] Issue still occurs even after testing on a new server
- [✔] I attempted testing for the issue with the vanilla client (no mods)
- [✔] Issue still occurs even after testing with the vanilla Minecraft client

**Comment by bigpresh** [comment]
Some of this sounds like the confusion over how deep claims go; claims always protect from the level the claim was made at all the way to the sky, but they do *not* extend all the way down in to the ground - they will extend down the value of the `ExtendIntoGroundDistance` config setting - which is set to 5 blocks in your config above.  *However*, they will automatically descend to protect blocks the owner places underneath them.  The rationale being to stop surface claims preventing people from mining deep underground, whilst also protecting the area if someone builds underground.

The docs under [claim-limits](/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#claim-limits) for `Claims.ExtendIntoGroundDistance` explains how it works.

**Comment by RoboMWM** [comment]
> 2.- The zone is fully protected in the whole Y-coordinate.

This can't be true given your config, please verify this.

As for an explanation on this behavior, bigpresh has provided that, along with where you can change this.

**Comment by GamerElit** [comment]
I will try that, thank you very much!
And the tests we did did protect everything from the start. Then if we extended it was the problem.
I used about 10 players to help me, some misinformation they could give me. We spent a long time testing that.

**Comment by rividall** [comment]
Actually i have the same issue. Claims go all the way down to bedrock with the default 5blocks in extendIntoGroundDistance.
Any updates on this?

**Comment by Jikoo** [comment]
> Actually i have the same issue. Claims go all the way down to bedrock with the default 5blocks in extendIntoGroundDistance. Any updates on this?

If you are using creative claims mode this is normal. Otherwise, please fill out the template.

Note that claims will extend down further as players build down further.

**Comment by Jikoo** [comment]
Actually, going to close this particular issue as it's pretty old and looks like it was probably the issue with subclaims not inheriting or not extending parent depth, which has since been fixed. If you're having a related issue, please open a new ticket.

Presumably fixed by #1598
